charts/IndicatorDropdown.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from "react"
+import AsyncSelect from "react-select/async"
+import { ValueType, SingleValueProps } from "react-select"
+import { bind } from "decko"
+
+import {
+    ExplorerViewContext,
+    ExplorerViewContextType
+} from "./ExplorerViewContext"
+import { observable, action, runInAction } from "mobx"
+import { Indicator } from "./Indicator"
+import { observer } from "mobx-react"
+
+export interface IndicatorDropdownProps {
+    placeholder: string
+    selectedId?: number
+    onChangeId: (id: number) => void
+}
+
+@observer
+export class IndicatorDropdown extends React.Component<IndicatorDropdownProps> {
+    static contextType = ExplorerViewContext
+    context!: ExplorerViewContextType
+
+    static defaultProps = {
+        placeholder: "Type to search..."
+    }
+
+    @observable.ref currentIndicator?: Indicator | null
+
+    componentDidMount() {
+        this.loadCurrentIndicator()
+    }
+
+    componentDidUpdate() {
+        this.loadCurrentIndicator()
+    }
+
+    // Since this component only takes IDs as parameters, we need to load the
+    // full indicator in order to display the title.
+    @action async loadCurrentIndicator() {
+        if (
+            this.props.selectedId &&
+            this.props.selectedId !==
+                (this.currentIndicator && this.currentIndicator.id)
+        ) {
+            const fetchedId = this.props.selectedId
+            const indicator = await this.context.indicatorStore.get(
+                this.props.selectedId
+            )
+            // this.props.selectedId could've changed during the await, need to
+            // make a second check.
+            if (fetchedId === this.props.selectedId) {
+                runInAction(() => {
+                    this.currentIndicator = indicator
+                })
+            }
+        } else if (!this.props.selectedId) {
+            this.currentIndicator = null
+        }
+    }
+
+    @bind onChange(indicator: ValueType<Indicator>) {
+        // The onChange method can return an array of values (when multiple
+        // items can be selected) or a single value. Since we are certain that
+        // we are not using the multi-option select we can force the type to be
+        // a single value.
+        this.props.onChangeId((indicator as Indicator).id)
+    }
+
+    @bind async loadOptions(input: string): Promise<Indicator[]> {
+        return await this.context.indicatorStore.search({
+            query: input
```

Comment:
Super-minor but when it makes sense, I like to name my variables the same as their keys to take advantage of the more succinct object syntax. If you name the param `query` then this could just be `.search({query})`

Reply:
:+1: I agree, changed